Lib/test/test_xml_etree.py

```diff
@@ -115,6 +120,75 @@ def test_all(self):
         support.check__all__(self, ET, names, blacklist=("HTML_EMPTY",))
 
 
+class ElementTree_Element_UnitTest(unittest.TestCase):
+
+    def setUp(self):
+        self.element_foo = ET.Element("foo")
+
+    def test_copy(self):
+        with self.assertWarns(DeprecationWarning):
+            element_foo2 = self.element_foo.copy()
+
+        self.assertIsNot(element_foo2, self.element_foo)
+        # A copy is made of the attrib list during the creation of an element.
+        self.assertIsNot(element_foo2.attrib, self.element_foo.attrib)
+
+        self.assertIs(element_foo2.tag, self.element_foo.tag)
+        self.assertIs(element_foo2.text, self.element_foo.text)
+        self.assertIs(element_foo2.tail, self.element_foo.tail)
+
+        self.assertEqual(element_foo2.tag, self.element_foo.tag)
+        self.assertEqual(element_foo2.attrib, self.element_foo.attrib)
+        self.assertEqual(element_foo2.text, self.element_foo.text)
+        self.assertEqual(element_foo2.tail, self.element_foo.tail)
+
+        for (child1, child2) in zip(self.element_foo, element_foo2):
+            self.assertIs(child1, child2)
```

Comment:
I dunno if this is a problem with the code coverage software or what, but it doesn't seem like this statement is getting hit by the coverage checker. Is it possible that one of these two has length zero?

I'll note that I personally prefer to use `itertools.zip_longest` for tests like this, since `zip` will just silently fail if one of the two iterators is a truncated version of the other - e.g. `all(x == y for x, y in zip((1, 2, 3), (1, 2, 3, 4, 5)))`.

Reply:
The reason the statement wasn't getting hit was because I forgot to actually check it.

And while the lists are expected to be of identical length, I agree that failing silently is not a good idea.

I've created a new commit that addresses these two issues, plus others.